### DISCUSSION TITLE
Why GLoot use custom types and script classes together?

### DISCUSSION THREAD

**Discussion by hellowin** [peter-kish/gloot#139] [Q&A] [answered]
Hi, I want to ask, why GLoot is using a [custom types](https://github.com/peter-kish/gloot/blob/master/addons/gloot/gloot.gd#L19) and [script classes](https://github.com/peter-kish/gloot/blob/master/addons/gloot/core/inventory_item.gd#L3) together?

I think we only need one type from the reference [here](https://docs.godotengine.org/en/stable/tutorials/best_practices/scenes_versus_scripts.html#named-types).

**Comment by peter-kish** [comment, marked as answer]
Yup, you're right, there's not reason to have both.
It's just something that I thought it was was necessary back when I was learning to make plugins and never bothered to fix it.
I will probably remove the custom types and stick to script classes in the next release.

**Reply by hellowin** [reply]
Thank you for your humble answer. I appreciate your work 👍
